tools-v2/pkg/cli/command/curvebs/status/client/client.go

```diff
@@ -0,0 +1,241 @@
+				}
+			}
+			results <- basecmd.MetricResult{
+				Addr:  m.Addrs[0],
+				Key:   key,
+				Value: value,
+				Err:   err,
+			}
+		}(metric)
+	}
+
+	clientsInfo := map[string]map[string]string{}
+	count := 0
+	var errs []*cmderror.CmdError
+	var recordAddrs []string
+	for res := range results {
+		if res.Err.TypeCode() != cmderror.CODE_SUCCESS {
+			index := slices.Index(recordAddrs, res.Addr)
+			if index == -1 {
+				errs = append(errs, res.Err)
+				recordAddrs = append(recordAddrs, res.Addr)
+			}
+		} else if _, ok := clientsInfo[res.Addr]; ok {
+			clientsInfo[res.Addr][res.Key] = res.Value
+		} else {
+			clientsInfo[res.Addr] = make(map[string]string)
+			clientsInfo[res.Addr][res.Key] = res.Value
+		}
+		count++
+		if count >= size {
+			break
+		}
+	}
+
+	// process type => version => addrs
+	clientsAddrs := map[string]map[string][]string{}
+	for addr, mp := range clientsInfo {
+		if _, ok := clientsAddrs[mp[PROCESS_CMD_KEY]]; !ok {
+			clientsAddrs[mp[PROCESS_CMD_KEY]] = make(map[string][]string)
+		}
+		if _, ok := clientsAddrs[mp[PROCESS_CMD_KEY]][mp[VERSION_KEY]]; !ok {
+			clientsAddrs[mp[PROCESS_CMD_KEY]][mp[VERSION_KEY]] = make([]string, 0)
+		}
+		clientsAddrs[mp[PROCESS_CMD_KEY]][mp[VERSION_KEY]] = append(clientsAddrs[mp[PROCESS_CMD_KEY]][mp[VERSION_KEY]], addr)
+	}
+
+	for process_type, mp := range clientsAddrs {
+		for version, addrs := range mp {
+			for _, addr := range addrs {
+				row := make(map[string]string)
+				row[cobrautil.ROW_TYPE] = process_type
+				row[cobrautil.ROW_VERSION] = version
+				row[cobrautil.ROW_ADDR] = addr
+				row[cobrautil.ROW_NUM] = strconv.Itoa(len(addrs))
+				cCmd.rows = append(cCmd.rows, row)
+			}
+		}
+	}
+
+	mergeErr := cmderror.MergeCmdErrorExceptSuccess(errs)
+	cCmd.Error = mergeErr
+	list := cobrautil.ListMap2ListSortByKeys(cCmd.rows, cCmd.Header, []string{
+		cobrautil.ROW_TYPE, cobrautil.ROW_VERSION, cobrautil.ROW_ADDR,
+	})
+	cCmd.TableNew.AppendBulk(list)
+	cCmd.Result = cCmd.rows
+
+	return nil
+}
+
+func (cCmd *ClientCommand) ResultPlainOutput() error {
+	return output.FinalCmdOutputPlain(&cCmd.FinalCurveCmd)
+}
+
+func (cCmd *ClientCommand) GetProcessNameFromProcessResp(process_resp *string) string {
+	if find := strings.Contains(*process_resp, kProcessNebdServer); find {
+		return kProcessNebdServer
+	} else if find := strings.Contains(*process_resp, kProcessPython); find {
+		return kProcessPython
+	} else if find := strings.Contains(*process_resp, kProcessQemu); find {
+		return kProcessQemu
+	} else {
+		return kProcessOther
+	}
+}
+
+func NewStatusClientCommand() *ClientCommand {
+	clientCmd := &ClientCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "client",
+			Short:   "get status of client",
+			Example: mdsExample,
+		},
+	}
+	basecmd.NewFinalCurveCli(&clientCmd.FinalCurveCmd, clientCmd)
+	return clientCmd
+}
+
+func GetClientStatus(caller *cobra.Command) (*interface{}, *tablewriter.Table, *cmderror.CmdError) {
```

Comment:
if no other command use this no need this func

Reply:
done